### PR TITLE
Feature(use-filter): add useFilter composable and documentation

### PR DESCRIPTION
**Pull request by mateuszRybczonek**
This PR adds a useFilter composable that can be used to filter datasets in tables.

For usage please have a look at the .mdx file or host console InvoicesPage
(https://github.com/Holo-Host/host-console-ui/blob/feature/invoices-filtering/src/pages/InvoicesPage.vue)

### CODE REVIEW

**Review by Paterick — APPROVED** [review]
This is really cool, thanks Mateusz! Also, thanks for the great documentation on the PR.

File: src/composables/useFilter.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+```
+EFilterTypes = {
+  string,
+  number
+}
+
+interface FilterCriterion {
+  key: string           // field name we want to filter by
+  type: EFilterTypes    // type of filter
+  minLength: number     // minimum length of the filter value we want this criterion to be apllied
```
Comment (robbiecarlton) [inline src/composables/useFilter.mdx]:
```suggestion
  minLength: number     // minimum length of the filter value we want this criterion to be applied
```

File: src/composables/useFilter.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export const EFilterTypes = {
+  string: 0,
+  number: 1
+}
+
+export function useFilter({ data, criteria }) {
+  const filterValue = ref('')
+  const filterIsActive = ref(false)
+
+  function onFilterChange({ value, isActive }) {
```
Comment (robbiecarlton) [inline src/composables/useFilter.js]:
```suggestion
  function setFilter({ value, isActive }) {
```

I think calling this `setFilter` more clearly communicates what it does. I read the example usage first, and it took me a second to understand what `onFilterChange` did.

**Review by robbiecarlton — APPROVED** [review]
one non blocking suggestion about naming, overall this is great, and awesome documentation! That's a great pattern.

**Pull request merged**